mflowgen/components/graph.py

```diff
@@ -107,19 +107,24 @@ def get_step( s, step_name ):
     return s._steps[ step_name ]
 
   def all_steps( s ):
-    return s._steps.keys()
+    return sorted( s._steps.keys() )
 
   # Edges -- incoming and outgoing adjacency lists
+  # Sort them for better debuggability / repeatability / causality
+
+  def sort_edges( s, edge_list ):
+    edge_list.sort(key=lambda x: x.src)
+    return edge_list
 
   def get_edges_i( s, step_name ):
     try:
-      return s._edges_i[ step_name ]
+      return s.sort_edges(s._edges_i[ step_name ])
     except KeyError:
       return []
 
   def get_edges_o( s, step_name ):
     try:
-      return s._edges_o[ step_name ]
+      return s.sort_edges(s._edges_0[ step_name ])
```

Comment:
This should be edges_o, not edges_0.

Reply:
Well that's disturbing. I had no idea that I had changed code that was not being tested :(
So then after I fixed the typo, I wrote a test that did exercise the code, and didn't particularly like what I saw, so changed the sort to use `dst` as the key instead of `src,` as that order seemed visually more pleasing.
I don't think it matters in the end, and any order is better than none I guess; but anyway I will give you a couple days to respond to this change, if you like, before I do the final merge.

Reply:
Looks great, feel free to merge!